src/ListSource.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_list_pages;
+
+use Drupal\Core\Plugin\PluginBase;
+use Drupal\search_api\Datasource\DatasourceInterface;
+
+/**
+ * List source object.
+ */
+class ListSource {
+
+  /**
+   * The id for the list source.
+   *
+   * @var string
+   */
+  protected $id;
+
+  /**
+   * The entity type.
+   *
+   * @var string
+   */
+  protected $entityType;
+
+  /**
+   * The bundle.
+   *
+   * @var string
+   */
+  protected $bundle;
+
+  /**
+   * The data source.
+   *
+   * @var \Drupal\search_api\Datasource\DatasourceInterface
+   */
+  protected $dataSource;
+
+  /**
+   * ListSource constructor.
+   *
+   * @param string $entity_type
+   *   The entity type.
+   * @param string $bundle
+   *   The bundle.
+   */
+  public function __construct(string $entity_type, string $bundle) {
+    $this->entityType = $entity_type;
+    $this->bundle = $bundle;
+    $this->id = 'list_facet_source' . PluginBase::DERIVATIVE_SEPARATOR . $entity_type . PluginBase::DERIVATIVE_SEPARATOR . $bundle;
+
+  }
+
+  /**
+   * Get available filters for the list source.
+   *
+   * @return array
+   *   The filters.
+   */
+  public function getAvailableFilters(): array {
```

Comment:
This method does not belong here on a value object if it needs to calculate the filters from a service.

Either you add the available filters when you instantiate the object (if you need to do this in different places you can use a central factory service) or simply move this method elsewhere like on the manager or whatever.